src/compile.js

```diff
@@ -2043,7 +2050,7 @@ Compiler.prototype.buildcodeobj = function (n, coname, decorator_list, args, cal
 
     // inject __class__ cell when running python3
     if (Sk.python3 && class_for_super) {
-        this.u.varDeclsCode += "$gbl.__class__=this." + class_for_super.v + ";";
```

Comment:
this is an odd change 😄

Reply:
It's because, in `co_fastcall` land, `this` is no longer the globals object; it's the `func` object.

Reply:
ah yes that makes sense

Reply:
This seems to be the change that is preventing it passing the tests.
For some reason this change got lost in the version travis is testing...